include/igraph_operators.h

```diff
@@ -40,6 +40,10 @@ __BEGIN_DECLS
 /* Graph operators                                    */
 /* -------------------------------------------------- */
 
+typedef enum {
+        IGRAPH_CARTESIAN_PRODUCT,
+} igraph_product_t;    
+
```

Comment:
Please put this in `igraph_constants.h`, add `= 0` for predictability, and name all values as `IGRAPH_PRODUCT_XXX`, i.e. `IGRAPH_PRODUCT_CARTESIAN`.